### PR TITLE
InsertWithOutput, InsertWithOutputInto for MSSQL.

### PR DESCRIPTION
**Pull request by sdanyliv**
Remaining tasks:

- [x]  Async versions
- [x]  Complete tests

### CODE REVIEW

File: Source/LinqToDB/Linq/Builder/InsertBuilder.cs [hunk 1]
```diff
@@ -123,18 +245,45 @@ protected override IBuildContext BuildMethodCall(ExpressionBuilder builder, Meth
...
+				switch (_insertType)
+				{
+					case InsertType.Insert:
+						QueryRunner.SetNonQueryQuery(query);
+						break;
+					case InsertType.InsertWithIdentity:
+						QueryRunner.SetScalarQuery(query);
+						break;
+					case InsertType.InsertOutput:
+						//TODO:
```
Comment (MaceWindu) [inline Source/LinqToDB/Linq/Builder/InsertBuilder.cs]: TODO what? :-)

Comment (sdanyliv) [reply Source/LinqToDB/Linq/Builder/InsertBuilder.cs]: Looks like obsolete comment.

File: Source/LinqToDB/Linq/Builder/InsertBuilder.cs [hunk 2]
```diff
@@ -2,19 +2,22 @@
...
+	using LinqToDB.Expressions;
 
 	class InsertBuilder : MethodCallBuilder
 	{
 		#region InsertBuilder
 
 		protected override bool CanBuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
 		{
-			return methodCall.IsQueryable("Insert", "InsertWithIdentity");
+			return methodCall.IsQueryable("Insert", "InsertWithIdentity", "InsertWithOutput", "InsertWithOutputAsync", "InsertWithOutputInto");
```
Comment (MaceWindu) [inline Source/LinqToDB/Linq/Builder/InsertBuilder.cs]: TODO: check if we need to add  InsertWithOutputIntoAsync here

File: Source/LinqToDB/LinqExtensions.Insert.cs [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+		/// Inserts records from source query into target table and returns newly created records.
+		/// </summary>
+		/// <typeparam name="TSource">Source query record type.</typeparam>
+		/// <typeparam name="TTarget">Target table record type.</typeparam>
+		/// <param name="source">Source query, that returns data for insert operation.</param>
+		/// <param name="target">Target table.</param>
+		/// <param name="setter">Inserted record constructor expression.
+		/// Expression supports only target table record new expression with field initializers.</param>
+		/// <returns>Enumeration of records.</returns>
+		public static IEnumerable<TTarget> InsertWithOutput<TSource,TTarget>(
```
Comment (MaceWindu) [inline Source/LinqToDB/LinqExtensions.Insert.cs]: This method miss Async version

**Comment by MaceWindu** [comment]
working on remote context support, it's a bit tricky...

**Pull request merged**